frontend/src/components/modals/settings/SettingsModal.tsx

```diff
@@ -24,6 +25,7 @@ interface SettingsProps {
 function SettingsModal({ isOpen, onOpenChange }: SettingsProps) {
   const { t } = useTranslation();
   const currentSettings = getSettings();
+  const defaultSettings = getDefaultSettings();
```

Comment:
This does not need to be on the top-level of the function, it is only used in the scope of your new `handleResetSettings` function.